amazon/ion/ioncmodule.c

```diff
@@ -543,6 +527,7 @@ iERR ionc_write_value(hWRITER writer, PyObject* obj, PyObject* tuple_as_sexp) {
 
         ION_DECIMAL decimal_value;
         IONCHECK(ion_decimal_from_string(&decimal_value, decimal_c_str, &dec_context));
+        // todo: don't we need to also free decimal_c_str?
```

Comment:
I don't believe so. `c_string_from_py` calls [PyUnicode_AsUTF8AndSize](https://python.readthedocs.io/en/latest/c-api/unicode.html#c.PyUnicode_AsUTF8AndSize), the documentation for which says "This caches the UTF-8 representation of the string in the Unicode object, and subsequent calls will return a pointer to the same buffer. **The caller is not responsible for deallocating the buffer.**"

Reply:
Thanks for reading what was a "come back to this later" note. :) Will check myself and likely remove.